CorsixTH/Lua/hospital.lua

```diff
@@ -722,18 +722,20 @@ end
 
 function Hospital:purchasePlot(plot_number)
   local map = self.world.map
-  if map.th:isParcelPurchasable(plot_number, self:getPlayerIndex()) and not self.world.ui.transparent_walls then
+  if map.th:isParcelPurchasable(plot_number, self:getPlayerIndex()) then
     local cost = not self.world.free_build_mode and map:getParcelPrice(plot_number) or 0
     if cost <= self.balance then
       self.world:setPlotOwner(plot_number, self:getPlayerIndex())
       table.insert(self.ownedPlots, plot_number)
-      -- Also make sure to apply transparency to the new walls, if required.
-      self.world.ui:applyTransparency()
+      if self:isPlayerHospital() and self.world.ui.transparent_walls then
```

Comment:
why would we only apply transparency to the new walls if it was the current player?

Reply:
Yes, fixed